ooniprobe/Model/NetworkMeasurement.mm

```diff
@@ -503,33 +311,38 @@ -(void)run{
 }
 
 -(void) run_test {
-    [super run_test];
-    mk::nettests::NdtTest()
-        .set_options("geoip_country_path", [geoip_country UTF8String])
-        .set_options("geoip_asn_path", [geoip_asn UTF8String])
-        .set_options("save_real_probe_ip", include_ip)
-        .set_options("save_real_probe_asn", include_asn)
-        .set_options("save_real_probe_cc", include_cc)
-        .set_options("no_collector", !upload_results)
-        .set_options("software_name", [@"ooniprobe-ios" UTF8String])
-        .set_options("software_version", [software_version UTF8String])
-        .set_verbosity(VERBOSITY)
-        .set_output_filepath([[self getFileName:@"json"] UTF8String])
-        .set_error_filepath([[self getFileName:@"log"] UTF8String])
-        .on_progress([self](double prog, const char *s) {
-            [self updateProgress:prog];
-        })
-        .on_log([self](uint32_t type, const char *s) {
-            #ifdef DEBUG
-            NSLog(@"%s", s);
-            #endif
-        })
-        .on_entry([self](std::string s) {
-            [self on_entry_ndt:s.c_str()];
-        })
-        .start([self]() {
-            [self testEnded];
-        });
+    //[super run_test];
+    mk::nettests::NdtTest test;
+    [super run_test:test];
```

Comment:
I would not call `run_test` the `super` method, because it does not actually run the test. I'd call it `init_common` or something that conveys the meaning that that is the common initialization path.